workshop/content/basics/templates/pseudo-parameters/_index.md

```diff
@@ -0,0 +1,224 @@
+
+You want to construct the resource ARN using pseudo parameters like the one in the following example. Let's say, for example, you have a sample parameter called `dbUsername` that you create in the `us-east-1` region, and in the AWS account `111122223333`; an example parameter ARN is constructed as in this example policy snippet:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ssm:GetParameters"
+            ],
+            "Resource": "arn:aws:ssm:us-east-1:111122223333:parameter/dbUsername"
+        }
+    ]
+}
+```
+Let's describe resources you need in your CloudFormation template.
+
+Let’s start by defining a [template parameter](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html) you will use as the value of the SSM parameter that you'll define in the template. Start by defining a template parameter called `DatabaseUsername` under the `Parameters` section; next, choose to reference this template parameter in the `Resources` section by using the `Ref` intrinsic function. Choose to copy content shown next, and paste it in the `pseudo-parameters.yaml` file, by appending it to the existing file content:
+
+
+```YAML
+Parameters:
+  DatabaseUsername:
+    AllowedPattern: ^[a-z0-9]{5,12}$
+    Type: String
+    Default: alice
+    Description: Value to be used with the dbUsername SSM parameter. The default value is set to 'alice', which users can override when creating a CloudFormation stack.
+
+```
+
+Next, describe an SSM parameter: choose to set its `Name` property to `dbUsername`, and use `Ref` to reference the `DatabaseUsername` template parameter in the `Value` property. Choose to append the following content to the existing file content:
+
+```YAML
+Resources:
+  BasicParameter:
+    Type: AWS::SSM::Parameter
+    Properties:
+      Name: dbUsername
+      Type: String
+      Value: !Ref DatabaseUsername
+      Description: SSM Parameter for database username.
+
+```
+
+Next, let’s define the IAM role and policy from where you want to reference the SSM parameter you defined above. Choose to copy content shown next, and paste it in the `Resources` section of the `pseudo-parameters.yaml` file by appending it to the existing file content:
+
+```yaml
+  DemoRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - lambda.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: /
+      Policies:
+        - PolicyName: ssm-least-privilege
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action: ssm:GetParameter
+                Resource: '*'
+
```

Comment:
```suggestion
```

Reply:
hi @ravsau - I am sorry, I did not mean to remove the 3 backticks, but only to indent them to the left.  Can you add them back here but with no leading spaces?